src/App.jsx

```diff
@@ -14,45 +19,88 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const getSortedGoods = (goods, { sortField, isReversed }) => {
+  const visibleGoods = [...goods];
+
+  if (sortField) {
+    visibleGoods.sort((item1, item2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return item1.localeCompare(item2);
+        case SORT_FIELD_LENGTH:
+          return item1.length - item2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversed ? visibleGoods.reverse() : visibleGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortedGoods = getSortedGoods(goodsFromServer,
+    { sortField, isReversed });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const resetSortField = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info',
+            { 'is-light': sortField !== SORT_FIELD_NAME })}
+          onClick={() => setSortField(SORT_FIELD_NAME)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success',
+            { 'is-light': sortField !== SORT_FIELD_LENGTH })}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning',
+            { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(reversed => !reversed)}
+        >
+          Reverse
+        </button>
+
+        {(sortField || isReversed) && (
```

Comment:
you can create separate variable like `isShowResetButton`
this is will be more readable
```suggestion
        const isShowResetButton = sortField || isReversed;
...
        {isShowResetButton && (
```